src/EditorFeatures/CSharpTest/Interactive/NavigateTo/InteractiveNavigateToTests.cs

```diff
@@ -550,9 +542,9 @@ void assertDescription(string label, string value)
             var source = "class SyllableBreaking {int GetKeyWord; int get_key_word; string get_keyword; int getkeyword; int wake;}";
             await TestAsync(source, async w =>
             {
-                var expecteditem1 = new NavigateToItem("get_keyword", NavigateToItemKind.Field, "csharp", null, null, MatchKind.Regular, null);
-                var expecteditem2 = new NavigateToItem("get_key_word", NavigateToItemKind.Field, "csharp", null, null, MatchKind.Regular, null);
-                var expecteditem3 = new NavigateToItem("GetKeyWord", NavigateToItemKind.Field, "csharp", null, null, MatchKind.Regular, true, null);
+                var expecteditem1 = new NavigateToItem("get_keyword", NavigateToItemKind.Field, "csharp", null, null, s_emptyCamelCaseNonContiguousPrefixPatternMatch_NotCaseSensitive, null);
+                var expecteditem2 = new NavigateToItem("get_key_word", NavigateToItemKind.Field, "csharp", null, null, s_emptyCamelCaseNonContiguousPrefixPatternMatch_NotCaseSensitive, null);
```

Comment:
I also found this suspicious, but confirmed this is what pattern matcher returns. I guess it's because get_key_word is not properly camel cased or it's just a quirky case.

        /// <summary>
        /// All camel-humps in the pattern matched a camel-hump in the candidate.  The first camel-hump
        /// in the pattern matched the first camel-hump in the candidate.  There was at least one gap in 
        /// the camel-humps in the candidate that were matched.
        ///
        /// Example: "CP" matching "CodeFixProviderService"
        /// Example: "cp" matching "CodeFixProviderService"
        /// Example: "CoProv" matching "CodeFixProviderService"
        /// </summary>
        CamelCaseNonContiguousPrefix,

Reply:
Yeah... this is likely something we could tweak in the future.  Hump-matching is really about matching pattern chunks to word chunks.  The issue, i think, is that we don't appropriately understand that get_key_word is 3 word chunks ('get', 'key', 'word').  i.e. we for hump matching, we should treat _ as word separators only, and not consider them as important for anything else.

Reply:
To be clear: that can happen in the future.  We've just never invested hugely in underscore cases as they're major in .Net/c#.  However, for the general pattern matcher (which can be used in many more cases), it's likely important to invest some more here.

Reply:
Yes, if the pattern has an underscore in it, it is being categorized as non contiguous. 
 And since we only have CamelCaseNonContiguousPrefix or CamelCaseNonContiguousSubstring, it is choosing one of those.  VS has the same behavior in their Pattern Matching as well.

Reply:
makes sense @olegtk do you want to track/open an issue for improving this here for the future?

Reply:
Sure, I filed https://devdiv.visualstudio.com/DevDiv/_workitems/edit/554833 bug to track this.